docs/solutions/logic-errors/command-usage-field-separation-2026-07-24.md

```
@@ -0,0 +1,124 @@
+  - dot-commands
+  - data-structure-design
+  - separation-of-concerns
+plan_ref: "decouple-command-usage-1168"
+---
+## Problem
+
+`Command.name` in the `COMMANDS` table doubled as both the dispatch/completion key and the `.help` usage-hint display. Commands whose names embedded usage syntax (e.g. `.rewind <n>`, `.edit message <n>`, `.info mcp [server]`) tab-completed to the literal usage string instead of a real command.
+
+## Symptoms
+
+- Typing `.rew` and pressing TAB completed to `.rewind <n> ` instead of `.rewind ` — the `<n>` placeholder appeared in the input buffer
```

markdownlint-cli2 (0.23.0)

[warning] Spaces inside code span elements

(MD038, no-space-in-code)